Content/Items/Gear/Weapons/Sword/BloodOath.cs

```diff
@@ -0,0 +1,186 @@
+﻿using System.Collections.Generic;
+using Microsoft.Xna.Framework.Graphics;
+using Newtonsoft.Json.Linq;
+using PathOfTerraria.Content.Projectiles.Melee;
+using PathOfTerraria.Core.Systems;
+using PathOfTerraria.Core.Systems.Affixes;
+using PathOfTerraria.Core.Systems.Affixes.ItemTypes;
+using ReLogic.Content;
+using Terraria;
+using Terraria.DataStructures;
+using Terraria.ID;
+using Terraria.Localization;
+
+namespace PathOfTerraria.Content.Items.Gear.Weapons.Sword;
+
+internal class BloodOath : Sword
+{
+	public override string Texture => $"{PathOfTerraria.ModName}/Assets/Items/Gear/Weapons/Sword/BloodOath";
+	public override float DropChance => 5f;
+	public override int ItemLevel => 1;
+	public override bool IsUnique => true;
+	public override string Description => Language.GetTextValue("Mods.PathOfTerraria.Items.BloodOath.Description");
+	public override string AltUseDescription => Language.GetTextValue("Mods.PathOfTerraria.Items.BloodOath.AltUseDescription");
+	protected override bool CloneNewInstances => true;
+
+	private readonly HashSet<int> _hitNpcs = [];
+
+	private bool _specialOn = false;
```

Comment:
Should we maybe add a "Toggle" system to the AltUseSystem - I feel like this won't be the only item that does this type of behavior

Reply:
The question is do we want that to be stored per item or per player. At the moment, you can switch off of Blood Oath and when you return the special will still be active without activating any other Blood Oath specials, instead of any alt-toggle being active.

Reply:
I can't imagine we would want more than 1 alt use toggled on at any given time. But maybe that isn't the case when we get to armor and stuff so I suppose this is fine